buf: 0.46.0 -> 0.49.0

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Bump buf to ~`0.48.2`~ `0.49.0`

Drop leaveDotGit (see below)
Slight cleanup
Added myself as a maintainer

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

06kellyjac commented:
Result of `nixpkgs-review pr 132426` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>buf</li>
  </ul>
</details>

raboof commented:
I get a different hash for this tag :thinking: 

```
From https://github.com/bufbuild/buf
 * tag               v0.48.2    -> FETCH_HEAD
Switched to a new branch 'fetchgit'
Enumerating objects: 1719, done.
Counting objects: 100% (1719/1719), done.
Compressing objects: 100% (1356/1356), done.
Writing objects: 100% (1719/1719), done.
Total 1719 (delta 386), reused 1321 (delta 0), pack-reused 0
Enumerating objects: 1719, done.
Nothing new to pack.
error: hash mismatch in fixed-output derivation '/nix/store/2fjccw2k7jmkksyzrwkiszhxbhzwgbbf-source.drv':
         specified: sha256-+l7CDqOq2S48tJqcmy60Yi6KaFnj1foQwA2rM2ISfTU=
            got:    sha256-sMWklO0g44Y5Apdz2bQA4PW0ZXsr9+Tvuw2Ce3t4WoQ=
```

SuperSandro2000 commented on pkgs/development/tools/buf/default.nix:
```diff
@@ -7,16 +7,16 @@
     rev = "v${version}";
-    sha256 = "sha256-5mjk31HuPNO/RhmMhIm3dAZAED/Kk33ObjC8KbPKRxk=";
+    sha256 = "sha256-+l7CDqOq2S48tJqcmy60Yi6KaFnj1foQwA2rM2ISfTU=";
     leaveDotGit = true; # Required by TestWorkspaceGit
```
Can we remove this and create a new git repo before running the tests?

06kellyjac replied:
Yeah im just going to patch out the one test that needs it

06kellyjac replied:
I did try just git init just in case but it also requires a commit so yeah patched it out. Git is still a check input though as it is used for another test with no issues

06kellyjac committed:
buf: 0.46.0 -> 0.49.0

Dropped leaveDotGit to avoid non-determinism issues
Skip test that requried .git
Slight cleanup
Add myself as a maintainer

06kellyjac commented:
Result of `nixpkgs-review pr 132426` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>buf</li>
  </ul>
</details>

raboof reviewed (approved):
seems to work (lint + break check don't report problems for a project where I don't expect any)
